src/Poseidon/CLI/Genoconvert.hs

```diff
@@ -98,10 +103,15 @@ convertGenoTo outFormat onlyGeno outPath removeOld outPlinkPopMode outZip pac =
             liftIO $ createDirectoryIfMissing True (dropTrailingPathSeparator x)
             return x
         Nothing -> return $ posPacBaseDir pac
-    let (outGabs, outSabs, outIabs) = (newBaseDir </> outGrel, newBaseDir </> outSrel, newBaseDir </> outIrel)
+    let outFilesAbs = map (newBaseDir </>) outFilesRel
+    let outFilesAbsTemp = do -- loop over files
+            f <- outFilesAbs
+            if drop (length f - 3) f == ".gz" then
```

Comment:
That's a bit hacky... Maybe it would be more easily readable with `takeExtension`?

Reply:
Yeah, absolutely, no idea what I was thinking. Rewritten with `takeExtension` as suggested. Thanks.